rig/machine_control/tests/test_struct_file.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+import tempfile
+
+from rig.machine_control.struct_file import read_struct_file, num
+
+
+@pytest.mark.parametrize(
+    "s, val",
+    [(b"10", 10),
+     (b"0x3a", 0x3a),
+     ])
+def test_num(s, val):
+    assert num(s) == val
+
+
+@pytest.fixture
+def temp_struct_file(data):
+    """Writes the test data into a temporary file to be tested against.
+    """
+    # Create the temporary file and write in the data
+    fp = tempfile.TemporaryFile()
```

Comment:
For what its worth, you may enjoy "StringIO" which makes a file-like out of a string.

https://docs.python.org/2/library/stringio.html

Also, doesn't this fixture leak temporary files?